ref-exchange/src/account_deposit.rs

```diff
@@ -26,25 +26,23 @@ pub struct AccountDeposit {
 impl AccountDeposit {
     /// Adds amount to the balance of given token while checking that storage is covered.
     pub fn add(&mut self, token: AccountId, amount: Balance) {
-        let prev_amount = *self.tokens.get(&token).unwrap_or(&0);
-        self.tokens.insert(token, prev_amount + amount);
-        assert!(
-            self.storage_usage() <= self.amount,
-            "ERR_INSUFFICIENT_STORAGE"
-        );
+        if let Some(x) = self.tokens.get_mut(&token) {
```

Comment:
using `get_mut` is an optimization - in most of the cases we don't need the tokens will be already registered.